rfqmsg/reject.go

```diff
@@ -0,0 +1,213 @@
+package rfqmsg
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/lightningnetwork/lnd/routing/route"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+const (
+	// Reject message type field TLV types.
+
+	TypeRejectID  tlv.Type = 0
+	TypeRejectErr tlv.Type = 1
+)
+
+func TypeRecordRejectID(id *ID) tlv.Record {
+	return tlv.MakePrimitiveRecord(TypeRejectID, id)
+}
+
+func TypeRecordRejectErr(rejectErr *RejectErr) tlv.Record {
+	sizeFunc := func() uint64 {
+		return 4 + uint64(len(rejectErr.ErrMsg))
+	}
+	return tlv.MakeDynamicRecord(
+		TypeRejectErr, rejectErr, sizeFunc,
+		rejectErrEncoder, rejectErrDecoder,
+	)
+}
+
+func rejectErrEncoder(w io.Writer, val any, buf *[8]byte) error {
+	if typ, ok := val.(**RejectErr); ok {
+		v := *typ
+
+		// Encode error code.
+		err := tlv.EUint8T(w, v.ErrCode, buf)
+		if err != nil {
+			return err
+		}
+
+		// Encode error message.
+		msgBytes := []byte(v.ErrMsg)
+		err = tlv.EVarBytes(w, msgBytes, buf)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+
+	return tlv.NewTypeForEncodingErr(val, "RejectErr")
+}
+
+func rejectErrDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
+	if typ, ok := val.(**RejectErr); ok {
+		var errCode uint8
+		err := tlv.DUint8(r, &errCode, buf, 4)
+		if err != nil {
+			return err
+		}
+
+		var errMsgBytes []byte
+		err = tlv.DVarBytes(r, &errMsgBytes, buf, l-4)
+		if err != nil {
+			return err
+		}
+
+		rejectErr := &RejectErr{
```

Comment:
Hmm, still need to finish getting through the diff, but rather than encode these messages as TLV blobs _within_ TLV blobs, they can just be normal wire messages. So you'd define the message as if it were in the `lnwire` pacakge.